ClassCastException in ApiResultConverterFactory when minify is enabled

### Describe the bug
Exception stack:

```
Caused by: java.lang.ClassCastException: java.lang.Class cannot be cast to java.lang.reflect.ParameterizedType
	at com.slack.eithernet.ApiResultConverterFactory.responseBodyConverter(ApiResult.kt:3)
	at retrofit2.Retrofit.nextResponseBodyConverter(Retrofit.java:6)
	at retrofit2.HttpServiceMethod.parseAnnotations(HttpServiceMethod.java:383)
	... 31 more
```

Only happens when `minifyEnabled` is true with Android Gradle Plugin version 7.0. 

### Requirements (place an `x` in each of the `[ ]`)**
* [x] I've read and understood the [Contributing guidelines](../CONTRIBUTING.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

### To Reproduce
Steps to reproduce the behavior:

Use build with `minfiyEnabled` set to true. Observe exception when creating the Retrofit API instance.

### Expected behavior
Exception should not happen regardless of `minifyEnabled` configuration.

#### Additional context
Add any other context about the problem here.

--- a/src/main/resources/META-INF/proguard/eithernet.pro
+++ b/src/main/resources/META-INF/proguard/eithernet.pro
@@ -1,2 +1,2 @@
-# Keep ApiResult
+# Keep ApiResult's generics or else R8 could strip them. We introspect these at runtime
 -keep,allowobfuscation,allowshrinking class com.slack.eithernet.ApiResult

--- a/src/main/resources/META-INF/proguard/eithernet.pro
+++ b/src/main/resources/META-INF/proguard/eithernet.pro
@@ -1,0 +1,2 @@
+# Keep ApiResult
+-keep,allowobfuscation,allowshrinking class com.slack.eithernet.ApiResult

